tests/on-event.test.ts

```diff
@@ -281,3 +282,114 @@ test('onEvent user properties are not passed for non-whitelisted events', async
+                ctx.status(400),
+                ctx.json({
+                    errors: [
+                        {
+                            type: "'external_id' or 'braze_id' or 'user_alias' is required",
+                            input_array: 'attributes',
+                            index: 0,
+                        },
+                    ],
+                })
+            )
+        })
+    )
+
+    // Create a meta object that we can pass into the setupPlugin and onEvent
+    const meta = {
+        config: {
+            brazeEndpoint: 'US-02',
+            eventsToExport: 'account created',
+            userPropertiesToExport: 'email',
+            importUserAttributesInAllEvents: 'Yes',
+        },
+        global: {},
+    } as BrazeMeta
+
+    await setupPlugin(meta)
+    try {
+        await onEvent(
+            {
+                event: '$identify',
+                timestamp: '2023-06-16T00:00:00.00Z',
+                properties: {
+                    $set: {
+                        email: 'test@posthog',
+                        name: 'Test User',
+                    },
+                    is_a_demo_user: true,
+                },
+                distinct_id: 'test',
+                ip: '',
+                site_url: '',
+                team_id: 0,
+                now: new Date().toISOString(),
+            },
+            meta
+        )
+    } catch (e) {
+        expect(e instanceof RetryError).toBeTruthy()
+        // @ts-ignore
+        expect(e.message).toEqual('Braze API error onEvent, retrying.')
+    }
+})
+
+test('Braze offline error (500 response)', async () => {
+    const mockService = jest.fn()
+
+    server.use(
+        rest.post('https://rest.iad-02.braze.com/users/track', (req, res, ctx) => {
+            const requestBody = req.body
+            mockService(requestBody)
+            return res(ctx.status(500), ctx.json({}))
+        })
+    )
+
+    // Create a meta object that we can pass into the setupPlugin and onEvent
+    const meta = {
+        config: {
+            brazeEndpoint: 'US-02',
+            eventsToExport: 'account created',
+            userPropertiesToExport: 'email',
+            importUserAttributesInAllEvents: 'Yes',
+        },
+        global: {},
+    } as BrazeMeta
+
+    await setupPlugin(meta)
+    try {
+        await onEvent(
+            {
+                event: '$identify',
+                timestamp: '2023-06-16T00:00:00.00Z',
+                properties: {
+                    $set: {
+                        email: 'test@posthog',
+                        name: 'Test User',
+                    },
+                    is_a_demo_user: true,
+                },
+                distinct_id: 'test',
+                ip: '',
+                site_url: '',
+                team_id: 0,
+                now: new Date().toISOString(),
+            },
+            meta
+        )
+    } catch (e) {
+        expect(e instanceof RetryError).toBeTruthy()
```

Comment:
It's possible that we never actually reach this point. Can you use `await expect(onEvent(...)).rejects...`